### DISCUSSION TITLE
How to convert <Editor/> to HTML with css

### DISCUSSION THREAD

**Discussion by shanthi-techie** [react-page/react-page#1153] [General] [answered]
I'm trying to convert react-page-editor to HTML using the following code.

const html = ReactDOMServer.renderToString(<Editor value={jsonConst} readOnly cellPlugins={cellPlugins} />); I'm also using material UI components inside the editor. When HTML is generated i dont see material UI css is generated. How to add CSS in the HTML ? 

Could anyone please help me with this ? I see inline css are getting added to HTML

**Comment by macrozone** [comment, marked as answer]
@shanthi-techie best ask in the material-ui community.

i would also like to know why you generate html from react-page? 

if its just for displaying, just use a SSR framework like nextjs. ReactPage works out-of-the-box with nextjs and others

**Reply by shanthi-techie** [reply]
I need to  generate pdf out of it. HTML will be stored in backend and the same will be used to generate PDF.

**Reply by macrozone** [reply]
ah ok, makes sense.

you could also just have an endpoint in nextjs or your ssr framework and point [pupeteer](https://blog.risingstack.com/pdf-from-html-node-js-puppeteer/)  to that endpoint to generate the pdf.

having said that, that o.c. requires to setup ssr with material-ui properly, see for example https://v4.mui.com/guides/server-rendering/#handling-the-request

that leads us to the solution that you could use with your approach:

```
import { ServerStyleSheets, ThemeProvider } from '@material-ui/core/styles';
const sheets = new ServerStyleSheets();

const html = ReactDOMServer.renderToString(
  sheets.collect( 
   <Layout><Editor .... /></Layout>
  )
)


const css = sheets.toString();



```

**Reply by shanthi-techie** [reply]
Definitely i will try these and update you.
Your inputs are so helpful, I really appreciate it. Thank you
